nx/lib/nx.ex

```diff
@@ -9167,6 +9167,84 @@ defmodule Nx do
     end
   end
 
+  @doc """
+  Finds the variance of a tensor.
+
+  ## Examples
+
+  iex> Nx.var(Nx.tensor([[1, 2], [3, 4]]))
+
+  #Nx.Tensor<
+  f32
+  1.25
+  >
+  """
+  @doc type: :element
+  @spec var(tensor :: Nx.Tensor.t()) :: Nx.Tensor.t()
+  def var(%Nx.Tensor{shape: shape} = tensor) do
+    total =
+      shape
+      |> Tuple.to_list()
+      |> Enum.reduce(fn val, acc -> acc * val end)
+
+    mean = mean(tensor)
+
+    tensor
+    |> subtract(mean)
+    |> power(2)
+    |> sum()
+    |> divide(total)
```

Comment:
We should document the exact formula, in particular that we divide by `N`. In statistical use cases it's often desired to divide by `N - 1` when calculating sample variance/stdev, since that gives an unbiased estimate.

For the record [numpy.var](https://numpy.org/doc/stable/reference/generated/numpy.var.html) has `ddof` option and divides by `N - ddof`, [torch.var](https://pytorch.org/docs/stable/generated/torch.var.html) has a more explicit `unbiased` option, while [tfp.stats.variance](https://www.tensorflow.org/probability/api_docs/python/tfp/stats/variance) always uses `N`.

Reply:
added. good call.